Modules/Filtering/Smoothing/include/itkMeanImageFilter.hxx

```diff
@@ -98,6 +104,56 @@ MeanImageFilter<TInputImage, TOutputImage>::GenerateDataInSubregion(
     ++outputIterator;
   }
 }
+
+template <typename TInputImage, typename TOutputImage>
+template <typename TPixelAccessPolicy, typename TValueType>
+void
+MeanImageFilter<TInputImage, TOutputImage>::GenerateDataInSubregion(
+  const TInputImage &                              inputImage,
+  TOutputImage &                                   outputImage,
+  const ImageRegion<InputImageDimension> &         imageRegion,
+  const std::vector<Offset<InputImageDimension>> & neighborhoodOffsets,
+  const VariableLengthVector<TValueType> *)
+{
+  const auto neighborhoodSize = static_cast<double>(neighborhoodOffsets.size());
+
+  using namespace Experimental;
+  auto neighborhoodRange = ShapedImageNeighborhoodRange<const InputImageType, TPixelAccessPolicy>(
+    inputImage, Index<InputImageDimension>(), neighborhoodOffsets);
+  auto outputIterator = ImageRegionRange<OutputImageType>(outputImage, imageRegion).begin();
+
+  InputRealType sumRef(inputImage.GetNumberOfComponentsPerPixel());
+
+  using PixelComponentType = typename NumericTraits<InputRealType>::ValueType;
+  sumRef.Fill(NumericTraits<PixelComponentType>::Zero);
+
+  // These temp variable are needed outside the loop for
+  // VariableLengthVectors to avoid memory allocations on a per-pixel
+  // basis.
+  const InputRealType &               zeroSum = sumRef;
+  InputRealType                       sum;
+  typename OutputImageType::PixelType out;
+
+  for (const auto & index : ImageRegionIndexRange<InputImageDimension>(imageRegion))
+  {
+    neighborhoodRange.SetLocation(index);
+
+    sum = zeroSum;
```

Comment:
Is this (`sum = zeroSum`) equivalent to `sum.Fill(NumericTraits<PixelComponentType>::Zero)`?

Reply:
Yup, I think that would remove the need for zeroSum, and sumRef. It's good to remove the hoops that had to be jumped through to get the code to work all the pixel types.